synapse/handlers/federation_event.py

```diff
@@ -897,6 +901,11 @@ async def _process_pulled_event(
                     backfilled=backfilled,
                 )
         except FederationError as e:
+            if backfilled:
```

Comment:
I think we decided that we'd *record* failures to get events regardless of `backfilled`?

Reply:
👍 Yes! I planned on making it more general when tackling https://github.com/matrix-org/synapse/issues/13700 but we can do it here (just poke again).

I aimed this PR at specifically addressing the backfill scenario. With follow-up PR's to add the additional scenarios so we can reason about the intricacies there.

Reply:
I think we may as well just remove the `if backfilled:` rather than adding and then removing them